### PR TITLE
API page on website.

### LINKED ISSUE #76: add api page
[Bountysource](https://www.bountysource.com/issues/26648068-add-api-page?utm_source=26648068&utm_medium=shield&utm_campaign=ISSUE_BADGE)

To tell people how to use our api, ref:
https://github.com/cdnjs/cdnjs#api

### PR DESCRIPTION
**Pull request by willyb321**
This PR adds an API page to the website, and also fixes #76 .

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
Would you please remove the twitter part?

**Comment by PeterDaveHello** [comment]
I think we should also provide some output example and links example including both human readable version and not.

**Comment by willyb321** [comment]
@PeterDaveHello I'll get that done now.

**Comment by PeterDaveHello** [comment]
@willyb321 Btw, would you please the screenshot here so the others can join the discussion easily?  Thanks.

**Comment by willyb321** [comment]
@PeterDaveHello [Here](http://i.imgur.com/zYSNom0.png) is a screenshot of the page currently.

**Comment by willyb321** [comment]
Alright I reckon I've got the examples down.

**Comment by PeterDaveHello** [comment]
it comes `You can "search" cdnjs via our API:` with the url not searching search, should adjustment this part.

File: templates/api.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		<br>
+		API will reture minified result by default, if you wanna have a human readable result, try
+		<code>
+		output=human
+		</code>
+		like so:
+		<br>
+		<pre><code>
+https://api.cdnjs.com/libraries?output=human
+https://api.cdnjs.com/libraries?search=[query]?output=human</code></pre>
```
Comment (PeterDaveHello) [inline templates/api.html]: The output indent looks strange here?

**Comment by willyb321** [comment]
> Can we just put some examples directly in the same page but not only the links?

Do you mean an iframe or an image or ?  

> it comes You can "search" cdnjs via our API: with the url not searching search, should adjustment this part.  

Fixed.  

> The output indent looks strange here?

Fixed  
@PeterDaveHello

**Comment by PeterDaveHello** [comment]
@willyb321 I mean just a code block as the same output from api :)

**Comment by willyb321** [comment]
@PeterDaveHello ah yes didn't think of that, will do.

**Comment by willyb321** [comment]
Ok, code blocks for api are done. [Here](http://i.imgur.com/fhn9RzU.png) is a snippet from the page.

**Comment by PeterDaveHello** [comment]
1. Looks like there are some unnecessary blank lines.
2. I think we can remove `(minified)` since that's what the API will do originally.
3. For the human-readable output, let's also apply hightlightjs syntax hightlight.
4. Please also display the whole url with query strings in examples without just links.

Looks like you cut some of the partial output in the codeblock, I'm not sure if this could be good for us, will think about it, we could work on the other parts first.

**Comment by PeterDaveHello** [comment]
We should also update the route in apiServer.js to redirect api.cdnjs.com to here.

**Comment by PeterDaveHello** [comment]
@willyb321 btw, please add (I added them for you already) `!` in the picture syntax for screenshot so that we can preview it directly, thanks.

**Comment by willyb321** [comment]
Ok will get that done in the morning.

**Comment by PeterDaveHello** [comment]
@willyb321 No problem!

**Comment by willyb321** [comment]
@PeterDaveHello Ok, I've done 1,2,3,4. Do tell me if theres anything else you need.

**Comment by willyb321** [comment]
Here is a screenshot for the current page.
image

File: templates/api.html [hunk 2]
```diff
@@ -1,5 +1,11 @@
+<head>
+<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/styles/default.min.css">
+<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/highlight.min.js"></script>
```
Comment (PeterDaveHello) [inline templates/api.html]: We have it in the layout.
https://github.com/cdnjs/new-website/blob/master/templates/layout.html#L163

BTW, please always use https and SRI.

**Comment by willyb321** [comment]
Ok will get that fixed. Just commuting atm.

**Comment by willyb321** [comment]
Ok I think thats what you wanted.

**Comment by willyb321** [comment]
@PeterDaveHello do you mean all the url's or just the ones in the example section?

**Comment by PeterDaveHello** [comment]
@willyb321 the examples, thanks

**Comment by willyb321** [comment]
@PeterDaveHello ok done.

**Comment by PeterDaveHello** [comment]
@willyb321 can we use some more precise query like https://api.cdnjs.com/libraries?search=ractive, so that we don't need to cut out the results in the codeblocks? Thanks.

**Comment by willyb321** [comment]
@PeterDaveHello ok, done.

**Comment by PeterDaveHello** [comment]
@willyb321 Thanks for your super prompt reply!

**Comment by PeterDaveHello** [comment]
@willyb321 Looks like the highlight isn't working?
BTW, in the examples, would you please move the <a> link tag from the "text" to the "url"? Thanks.

**Comment by PeterDaveHello** [comment]
I would like to update the heading part to:

--

You can query cdnjs via our API as below

Without any query parameters it will return the name and main file URL of every library on cdnjs:
`https://api.cdnjs.com/libraries`

To search, use: 
`https://api.cdnjs.com/libraries?search=[query]`

**Comment by PeterDaveHello** [comment]
> API will reture minified result by default, if you wanna have a human readable result, try `output=human `like so: 
->
> API will reture minified result by default, if you wanna have a human readable result, try `output=human` like so:

**Comment by PeterDaveHello** [comment]
@PeterDaveHello For the example output been cut out, please add notes about that so that user won't feel strange :0

**Comment by willyb321** [comment]
@PeterDaveHello ok, done!

**Comment by PeterDaveHello** [comment]
@willyb321 Since not all of them?

**Comment by willyb321** [comment]
@PeterDaveHello if you are talking about the cut out, I ended up fixing that so that I didn't need to cut out anything.

**Comment by PeterDaveHello** [comment]
@willyb321 and the links and highlight :)

**Comment by willyb321** [comment]
@PeterDaveHello oh, oops!

**Comment by willyb321** [comment]
@PeterDaveHello Ok, I've got that done. Sorry about the slow reply, fell asleep lol.

**Comment by PeterDaveHello** [comment]
@willyb321 no worry at all, don't need to be nervous :)

**Comment by PeterDaveHello** [comment]
The url, fields block, should we use `<code>` there? I thought the hightlight apply not suitable style to them.

**Comment by willyb321** [comment]
@PeterDaveHello I reckon its fine as is.

**Comment by PeterDaveHello** [comment]
The link part in example, actually I hope the `<a>` for link only apply to the url part inside `(`&`)`

**Comment by PeterDaveHello** [comment]
> @willyb321 For the example output been cut out, please add notes about that so that user won't feel strange :0

This part I didn't see the update yet.

**Comment by willyb321** [comment]
@PeterDaveHello ok so something like this ? 
image

**Comment by PeterDaveHello** [comment]
@willyb321 
> Specific Library Assets (https://api.cdnjs.com/libraries?search=mustcache)

**Comment by willyb321** [comment]
@PeterDaveHello ok thats all done. Here's an image 
image

**Comment by PeterDaveHello** [comment]
@willyb321 Thanks

**Comment by willyb321** [comment]
@PeterDaveHello ok so is that all you need?

**Comment by PeterDaveHello** [comment]
@willyb321 the indent in human readable result is wrong.

**Comment by willyb321** [comment]
@PeterDaveHello ok that should be better.

File: templates/api.html [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+</code></pre>
+<br>
+The API is served over Cloudflare with a hour expiry for requests.
+</p><br>
+<h1>Examples</h1>
+All Libraries (<a>https://api.cdnjs.com/libraries</a>) (Cut down a bit)<br>
+<pre><code class="json">{"results":[{"name":"1140","latest":"https://cdnjs.cloudflare.com/ajax/libs/1140/2.0/1140.min.css"},{"name":"1000hz-bootstrap-validator","latest":"https://cdnjs.cloudflare.com/ajax/libs/1000hz-bootstrap-validator/0.11.6/validator.min.js"},{"name":"10up-sanitize.css","latest":"https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/4.1.0/sanitize.min.css"},{"name":"16pixels","latest":"https://cdnjs.cloudflare.com/ajax/libs/16pixels/0.1.8/16pixels.min.css"},
+</code></pre>
+<br>
+All Libraries (Human Readable) (<a>https://api.cdnjs.com/libraries?output=human</a>) (Cut down a bit)<br>
```
Comment (PeterDaveHello) [inline templates/api.html]: this link doesn't work at all ... don't we need `href` in the `<a>` tag?

Comment (PeterDaveHello) [inline templates/api.html]: BTW, let's use a new line for the link, and remove `(`, `)`, so it'll look like:

All Libraries Human Readable output (Cut down a bit in the example)
https://api.cdnjs.com/libraries?output=human

**Comment by willyb321** [comment]
@PeterDaveHello Ok, that should now be fixed.

**Comment by willyb321** [comment]
Here is a pic

**Comment by PeterDaveHello** [comment]
@willyb321 thanks, hope this is the final fix:
The example output in All Libraries part, would you please use the same content for human readable part?

BTW, would you please squash/fixup these commits into a single commit? Thanks.

**Comment by willyb321** [comment]
@PeterDaveHello ok its squashed now, and the example is the same.

**Comment by PeterDaveHello** [comment]
Awesome, @willyb321 thank you for your patience and contribution!

**Pull request merged**